packages/swapper/src/swappers/cow/getCowSwapTradeQuote/getCowSwapTradeQuote.ts

```diff
@@ -126,18 +117,18 @@ export async function getCowSwapTradeQuote(
       feeData: {
         fee,
         chainSpecific: {
-          estimatedGas: feeData.chainSpecific.gasLimit,
+          estimatedGas: APPROVAL_GAS_LIMIT,
```

Comment:
It seems like we would want to use the `feeData` gasLimit still here, so we are returning gas info for either making the trade, or performing the approval as already described by `approvalFee`.

It does make me think that our call to `getFeeData` is incorrect as it doesn't include the `appData` from the quote response to calculate the accurate gasLimit given the `inputData` required to make the trade (assuming that is what `appData` is when returned)

Reply:
@kaladinlight some explanation on this : I did not include `appData` in the call to `getFeeData` because I would always get an error response while doing so. As a consequence, the `gasLimit` received from our `getFeeData` is probably not correct (insufficient to perform an approval from my blind testing), hence why I hardcoded this `APPROVAL_GAS_LIMIT`.
Not sure If I should spend time trying to bring back `appData` into `getFeeData` call.

Reply:
At minimum, please add a comment if as to why the hard code that happens to be enough to cover the trade. I would ultimately like to understand the error case with the `appData` being passed to `getFeeData` so we can properly calculate these fees. Do you have tx reference I can look at or the returned `appData` on hand you can send my way and I can take a look?

Reply:
@chxash this looks good for accurately estimating gas for approvals. i did a little more digging and the missing piece of info for myself is that it doesn't look like there are actually any network fees for the swapping through their protocol due to their fees being calculated into the actual trade. can you confirm this?

In that case, it seems like we don't really even need the anything but the `feeData.fee` and `feeData.chainSpecific.approvalFee`. Regardless, I think it is good for us to accurately estimate fees where we can so having the approval calculation is a good improvement.

Reply:
> @chxash this looks good for accurately estimating gas for approvals. i did a little more digging and the missing piece of info for myself is that it doesn't look like there are actually any network fees for the swapping through their protocol due to their fees being calculated into the actual trade. can you confirm this?

Exactly yes, sorry for not mentionning it earlier